xdsl/dialects/vector.py

```diff
@@ -54,4 +54,99 @@ def get(vector: Operation | SSAValue, ref: Operation | SSAValue,
         return Store.build(operands=[vector, ref, indices])
 
 
-Vector = Dialect([Load, Store], [])
+@irdl_op_definition
+class Broadcast(Operation):
+    name = "vector.broadcast"
+    source: Annotated[Operand, AnyAttr()]
+    vector: Annotated[OpResult, VectorType]
+
+    def verify_(self):
+        if self.source.typ != self.vector.typ.element_type:
+            raise VerifyException(
+                "Source operand and result vector must have the same element type."
+            )
+
+    @staticmethod
+    def get(source: Operation | SSAValue) -> Broadcast:
+        return Broadcast.build(operands=[source],
+                               result_types=[
+                                   VectorType.from_type_and_list(
+                                       SSAValue.get(source).typ)
+                               ])
+
+
+@irdl_op_definition
+class FMA(Operation):
+    name = "vector.fma"
+    lhs: Annotated[Operand, VectorType]
+    rhs: Annotated[Operand, VectorType]
+    acc: Annotated[Operand, VectorType]
+    res: Annotated[OpResult, VectorType]
+
+    def verify_(self):
+        res_type = self.res.typ.element_type
+        res_shape = self.res.typ.get_shape()
+
+        if res_type != self.lhs.typ.element_type or res_type != self.rhs.typ.element_type or res_type != self.acc.typ.element_type:
+            raise VerifyException(
+                "Result vector type must match with all source vectors.")
+
+        if res_shape != self.lhs.typ.get_shape(
+        ) or res_shape != self.rhs.typ.get_shape(
+        ) or res_shape != self.acc.typ.get_shape():
+            raise VerifyException(
+                "Result vector shape must match with all source vector shapes."
+            )
+
+    @staticmethod
+    def get(lhs: Operation | SSAValue, rhs: Operation | SSAValue,
+            acc: Operation | SSAValue) -> FMA:
+        return FMA.build(operands=[lhs, rhs, acc],
+                         result_types=[
+                             VectorType.from_type_and_list(
+                                 SSAValue.get(lhs).typ.element_type)
+                         ])
+
+
+@irdl_op_definition
+class Maskedload(Operation):
+    name = "vector.maskedload"
+    memref: Annotated[Operand, MemRefType]
```

Comment:
Why did you plan to rename this? All other names (besides `passthrough`) seem to be the same as in MLIR. I don't think it matters, just wondering...

Reply:
`base` (name used in `MLIR`) didn't made much sense to me here. Besides, this is named in a similar manner at other [places](https://github.com/xdslproject/xdsl/blob/53420dee071fd5d4905b5a464428171ef4ecf977/xdsl/dialects/memref.py#L59) in the repository.